pkg/resources/model/reconciler.go

```diff
@@ -417,3 +417,14 @@ func jsonFieldName(f reflect.StructField) string {
 	}
 	return f.Name
 }
+
+func hasIntersection(a, b []string) bool {
```

Comment:
[nit]
If the slices have many element, then maybe a `map[string]struct{}` would be better here, but let's cross that bridge once we get there.